### PR TITLE
fix(agentd): use custom idle timeout annotation for garbage collection

### LINKED ISSUE #342: agentd: custom idle timeout annotation is ignored by garbage collector
/kind bug

**What happened**:
When configuring a custom `sessionTimeout` or `maxSessionDuration` on a sandbox runtime (e.g. an `AgentRuntime` or `CodeInterpreter`), the `workloadmanager` successfully generates the Sandbox and sets the `runtime.agentcube.io/idle-timeout` annotation to the requested custom duration (e.g., `"5m"` or `"30m"`). 

However, the `agentd` garbage collection daemon completely ignores this custom annotation. In `pkg/agentd/agentd.go` inside the `Reconcile` loop, the Reconciler calculates sandbox expiration using the hardcoded `SessionExpirationTimeout` constant which is strictly set to `15 * time.Minute`:
```go
expirationTime := lastActivity.Add(SessionExpirationTimeout)
```
Consequently, all custom idle timeouts are completely bypassed, and every idle sandbox is garbage-collected exactly at 15 minutes.

**What you expected to happen**:
The `agentd` garbage collection daemon should check the `runtime.agentcube.io/idle-timeout` annotation on the Sandbox metadata. If the annotation exists and is a valid duration, it should parse and use that duration to compute the expiration time. It should fall back to the default `SessionExpirationTimeout` (15 minutes) only if the annotation is missing or invalid.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a `CodeInterpreter` or `AgentRuntime` resource with a custom session timeout (e.g., `5m`).
2. Invoke the sandbox to trigger creation. Note that the sandbox is created with the `runtime.agentcube.io/idle-timeout: "5m"` annotation.
3. Stop sending any requests to the sandbox so that it becomes idle.
4. Observe that `agentd` does not clean up the sandbox at 5 minutes, but instead cleans it up at exactly 15 minutes.

**Anything else we need to know?**:
This is a functional bug that prevents proper session lifecycle management for environments that need shorter timeouts (to save resources) or longer timeouts (for interactive user tasks).

**Environment**:

- agentcube version: v0.1.0 (main branch)
- Kubernetes version: Any supported K8s cluster (e.g. kind/v1.31)

### PR DESCRIPTION
**Pull request by hrmn3s**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
This PR fixes a bug where `agentd` ignores the custom `runtime.agentcube.io/idle-timeout` annotation set on a Sandbox. Previously, the garbage collection reconciliation loop used a hardcoded `15 * time.Minute` value for every sandbox. 

With this change, `agentd` successfully parses the Sandbox's custom idle timeout from the metadata annotations and correctly honors shorter or longer custom session durations. It will fall back to the default `15 * time.Minute` if the annotation is missing or improperly formatted.

It also introduces new unit test coverage inside `agentd_test.go` to explicitly verify custom short timeouts, custom long timeouts, and invalid timeout fallback paths.

**Which issue(s) this PR fixes**:
Fixes #342

**Special notes for your reviewer**:
* All changes covered by unit tests.

**Does this PR introduce a user-facing change?**:

```release-note
Fixes an issue where custom Sandbox idle timeouts (`runtime.agentcube.io/idle-timeout`) were ignored by the garbage collector and always fell back to 15 minutes.
```

### CODE REVIEW

**Comment by hrmn3s** [comment]
Hi @acsoto, @YaoZengzeng

I have updated this PR to completely streamline its scope and address the automated review feedback.
I would really appreciate it if you could take a look and share your thoughts. Thank you for your guidance and time!

File: pkg/agentd/agentd.go [hunk 1]
```diff
@@ -58,6 +58,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 
 		expirationTime := lastActivity.Add(SessionExpirationTimeout)
```
Comment (acsoto) [inline pkg/agentd/agentd.go]: Should this fallback use `workloadmanager.DefaultSandboxIdleTimeout` instead of the separate `SessionExpirationTimeout` constant? The annotation producer and agentd fallback are meant to represent the same default, so keeping two constants can drift.

**Review by hzxuzhonghu — COMMENTED** [review]
Clean implementation with proper fallback behavior.

`pkg/agentd/agentd.go`
- Line 62-67: Good defensive coding - the `err == nil && customTimeout > 0` guard correctly rejects invalid, zero, and negative durations. Consider logging a warning when the annotation is present but unparseable, so operators can spot misconfigurations without reading the source.

`pkg/agentd/agentd_test.go`
- Test coverage is thorough: covers short timeout (deleted), long timeout (requeued), invalid string, negative, and zero values. Good edge case coverage.

LGTM.

File: pkg/agentd/agentd.go (hunk 1 above)
Comment (hrmn3s) [reply pkg/agentd/agentd.go]: I've just pushed a commit that will fix this.
Thanks!

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm
